src/sas/sascalc/fit/BumpsFitting.py

```diff
@@ -310,7 +326,19 @@ def fit(self, msg_q=None,
                     param.value = val
 
         # Propagate correlated uncertainty through constraints.
-        problem.setp_hook()
+        # Hack to deal with linked constraints, e.g. M2=M1, M3=M2, M4=M3, ...
+        # TODO: Find better way of handling this?
+        MAX_ITER = 10
+        iter = 0
+        while not problem.setp_hook._allComputedParamsUncertaintiesDefined():
+            iter += 1
+            if iter > MAX_ITER:
+                logging.warn(f"Handling of uncertainties of constrained parameters "
+                            f"exited after {MAX_ITER} iterations.\n"
+                            f"This could be due to a too complex constraining scheme "
+                            f"or a very long chain of linked constraints.")
+                break
+            problem.setp_hook()
```

Comment:
The code in `sascalc.fit.expression` determines an expression evaluation order required to propagate the constraints through the various expressions. It should work equally well for propagating the uncertainties.

The fact that you need this loop suggests a deeper problem with the expression evaluation order, meaning that the constraints will not be consistent within the model. Fix the evaluation order in `_compile_constraints` and the uncertainty propagation should happen automatically.

Reply:
Thanks for the hint. The order of the constraints seems to be correctly evaluated, but the order.reverse() command at the end of the function seems to spoil it. By commenting it out everything seems to work fine. I don't know if there could be some special case where this reverse command was needed. From my side, I did several tries linking the constraints in different ways and found that without the reverse everything works as expected and the uncertainties for all the parameters are well resolved.